### PR TITLE
Explicitly decide whether to  use TLS in mailer's configuration

### LINKED ISSUE #4820: Mail module should add an option to explicitly decide using tls or not.
Dear All,
    in the module   gitea/modules/mailer/mailer.go,  there is a fragment to decide it using tls or not.
        isSecureConn := false
	// Start TLS directly if the port ends with 465 (SMTPS protocol)
	if strings.HasSuffix(port, "465") {
		conn = tls.Client(conn, tlsconfig)
		isSecureConn = true
	}
It says that the port should suffix with '465'. we have a problem it, because we use a smtp which using the port  946 for tls connect.  So I suggest add an option in the mail config  to support tls connect explicitly.
Thanks very much.

### PR DESCRIPTION
**Pull request by adelowo**
Fixes https://github.com/go-gitea/gitea/issues/4820

### CODE REVIEW

**Review by lunny — APPROVED** [review]

**Comment by lafriks** [comment]
It should probably use default depending on port set to not break compatibility

**Review by techknowlogick — APPROVED** [review]

File: modules/mailer/mailer.go [hunk 1]
```diff
@@ -122,11 +122,10 @@ func (s *smtpSender) Send(from string, to []string, msg io.WriterTo) error {
 	}
 	defer conn.Close()
 
-	isSecureConn := false
+	isSecureConn := opts.IsTLSEnabled
 	// Start TLS directly if the port ends with 465 (SMTPS protocol)
-	if strings.HasSuffix(port, "465") {
```
Comment (techknowlogick) [inline modules/mailer/mailer.go:127]: Perhaps have the logic be
```
isSecureConn =  (port =465) || TLSEnabled
```
for backwards compatibility per @lafriks suggestion?

**Review by techknowlogick — CHANGES_REQUESTED** [review]

**Comment by adelowo** [comment]
Sorry for not updating this on time. I have been a little busy this past 2 weeks.

I would be updating my other PRs before the start of next week

**Comment by techknowlogick** [comment]
@adelowo no worries. We appreciate the time you are able to give.

**Review by techknowlogick — APPROVED** [review]

**Pull request merged**